internal/exec/engine.go

```diff
@@ -446,11 +445,11 @@ func (e *Engine) fetchReferencedConfig(cfgRef types.Resource) (types.Config, err
 	return cfg, nil
 }
 
-func SignalNeedNet() error {
-	if err := executil.MkdirForFile(needNetPath); err != nil {
+func (e Engine) SignalNeedNet() error {
```

Comment:
I'm not sure what golang convention is, though ISTM it makes more sense to default to pointer receivers when operating on large structs to avoid copying a bunch of stuff just for simple read-only methods like these. Ideally we'd be able to declare the mutability intention in the signature like Rust.

Though actually, `Engine` has pointer members and AFAIK golang doesn't do a deep copy so even by value it's not really providing any assurance of immutability.

Obviously doesn't matter big picture and this is called on the way out anyway. So more of an aside discussion. :)

Reply:
Ah, yeah, `logReport()` used a value receiver so I did too, but the other methods do not.  I agree with your reasoning.